File: Strivers-A2Z/S3-Arrays/Easy/Largest_element_in_an_array.py
# Date: 1 Jan 2025
# Link: https://bit.ly/3Pld280
# Name: Largest Element in Array

from typing import List


class Solution:
    def largest(self, arr):
        max = arr[0]
        for i in range(len(arr)):
            max = arr[i] if (max < arr[i]) else max
        return max


obj = Solution()
nums = [1, 8, 7, 56, 90]
# nums = [5, 5, 5, 5]
# nums = [10]
print(obj.largest(nums))

File: Strivers-A2Z/Arrays/Easy/Largest_element_in_an_array.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    int largest(vector<int> &nums)
    {
        int largest = nums[0];
        for (size_t i = 1; i < nums.size(); i++)
            if (nums[i] > largest)
                largest = nums[i];
        return largest;
    }
};

int main()
{
    int n;
    cin >> n;
    vector<int> nums(n);
    for (int &x : nums)
        cin >> x;
    Solution obj;
    cout << obj.largest(nums);
    return 0;
}